packages/react/src/analytics/integrations/GTM/eventsMapper.js

```diff
@@ -5,63 +5,326 @@
 
 import { eventTypes, pageTypes } from '@farfetch/blackout-core/analytics';
 import { getEventProperties, getProductData } from './utils';
+import { isObject, round, snakeCase } from 'lodash';
 import get from 'lodash/get';
 
 const noPropertiesMappedFn = () => ({});
 
+/**
+ * Returns the total event value for GTM ecommerce events.
+ *
+ * @param {object} eventProperties - Properties from a track event.
+ * @param {Array} items - Items contained on event tracking.
+ *
+ * @returns {number} Event total value calculated.
+ */
+const getEventTotalValue = (eventProperties, items) => {
+  // There could be cases where the client is not using the bag middleware and wants to pass a value.
+  if (typeof eventProperties.value === 'number') {
+    return eventProperties.value;
+  }
+
+  const value = items?.reduce((acc, item) => {
+    const lineValue =
+      (get(item, 'price', 0) - get(item, 'discount', 0)) *
+      get(item, 'quantity', 1);
```

Comment:
why are you using the get here, just asking to learn?

What do you think about using an Object with these values? 
like it:

```typescript

const { price, discount, quantity } = item ?? {
  price: 0,
  discount: 0,
  quantity: 1,
};

const lineValue = (price - discount) * quantity;

```

it is closest to the vanilla JS, avoiding using a library.

Reply:
That will only work if item is undefined/null but we need to account for the cases where the item is set but some of the properties are not set (price, discount and quantity).